src/main/java/software/amazon/smithy/lsp/SmithyInterface.java

```diff
@@ -29,14 +32,31 @@ private SmithyInterface() {
 
   /**
    * @param path File of single-file model.
+   * @param workspaceRoot Workspace
    * @return Returns either an Exception, or the ValidatedResult of a model.
    */
-  public static Either<Exception, ValidatedResult<Model>> readModel(File path) {
-
+  public static Either<Exception, ValidatedResult<Model>> readModel(File path, File workspaceRoot) {
     try {
-      return Either.forRight(Model.assembler().discoverModels().addImport(path.getAbsolutePath()).assemble());
+      ModelAssembler assembler = Model.assembler();
+      assembler.discoverModels();
+      // If the workspace root has been set, get all the Smithy model files available.
+      if (workspaceRoot != null) {
+        List<File> modelFiles = getModelFiles(workspaceRoot);
+        modelFiles.forEach(f -> assembler.addImport(f.getAbsolutePath()));
+      } else {
+        // Fall back to the single model file.
+        assembler.addImport(path.getAbsolutePath());
+      }
+      return Either.forRight(assembler.assemble());
     } catch (Exception e) {
       return Either.forLeft(e);
     }
   }
+
+  private static List<File> getModelFiles(File root) {
+    Collection<File> smithyModelFiles = FileUtils.listFiles(root, new String[]{"smithy"}, true);
+    // Filter out any any model files that have been generated during a previous build.
+    return smithyModelFiles.stream().filter(f -> !f.getAbsolutePath().contains(root.getAbsolutePath() + "/build"))
+            .collect(Collectors.toList());
+  }
```

Comment:
I'm not sure this is the right approach. You could be pulling in a ton of files that aren't necessarily compatible with each other. For instance, projections could have `imports` with conflicting trait or even shape definitions.

What I would expect is that you travel up through directories from the current file's dir up to the workspace root searching for the first `smithy-build.json` and then run the same logic that the gradle plugin uses to load it. There's still potential questions about how to treat projections, but using the source projection will be the best bet. Maybe later some option to switch which projection you're looking at could be added.